cloudflare/data_source_zone.go

```diff
@@ -0,0 +1,94 @@
+package cloudflare
+
+import (
+	"fmt"
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"log"
+	"regexp"
+)
+
+func dataSourceCloudflareZone() *schema.Resource {
```

Comment:
I'm a little uncertain on whether a data source for a zone is needed as I would have expected that should you need the details, you would have defined the resource already.

Reply:
@jacobbednarz, create question. I should have put some context around this in the PR text. Actually we don't plan to manage the `zone` resource in the same terraform project. We prefer to delegate permission to that important resource to another project, which is only accessible to a team with a different set of credentials. This is why i thought the datasource would be so useful. It's how we currently manage our route53, using a [datasource](https://www.terraform.io/docs/providers/aws/d/route53_zone.html) to lookup the `zone_id` and then passing that when creating related resources

Reply:
Very interesting (and sensible!) use case. I'm 👌 with this approach given the context.

Reply:
Love the idea of Zone data source, in fact I've been asked for the same thing internally (a list of zones currently added to an account), but let's pull this out of this PR and create a separate PR.

1 PR == 1 feature